statsd/statsd.go

```diff
@@ -271,18 +293,23 @@ func (c *Client) Flush() error {
 	if c == nil {
 		return nil
 	}
-	c.Lock()
-	defer c.Unlock()
-	return c.flushLocked()
+	for _, b := range c.buffers {
+		func() {
+			b.Lock()
+			defer b.Unlock()
```

Comment:
can avoid defer here.

Reply:
Do you mean lock() -> flushLocked() -> unlock? We chose to follow the defer pattern for consistency. Are you worried it's incorrect or non-performant?

Reply:
I agree, we should just get rid of this defer